torch/distributed/utils.py

```diff
@@ -19,22 +19,36 @@ def to_map(obj):
         if isinstance(obj, torch.Tensor):
             if obj.device == torch.device("cuda", target_gpu):
                 return (obj,)
+            elif hasattr(torch, "xpu") and obj.device == torch.device("xpu", target_gpu):
+                return (obj,)
             if not use_side_stream_for_tensor_copies:
                 return (obj.to(target_gpu),)
             else:
                 # Perform CPU -> GPU copies in a background stream. This code is
                 # motivated from similar logic in torch/nn/parallel/_functions.py
                 stream = _get_stream(target_gpu)
-                with torch.cuda.stream(stream):
-                    output = obj.to(target_gpu)
-                # synchronize with the copy stream
-                with torch.cuda.device(target_gpu):
-                    current_stream = torch.cuda.current_stream()
-                    # Sync the current stream with the copy stream
-                    current_stream.wait_stream(stream)
-                    # Ensure tensor memory is not reused until work on
-                    # main stream is complete
-                    output.record_stream(current_stream)  # type: ignore[arg-type]
+                if torch.cuda.is_available():
+                    with torch.cuda.stream(stream):
+                        output = obj.to(target_gpu)
+                    # synchronize with the copy stream
+                    with torch.cuda.device(target_gpu):
+                        current_stream = torch.cuda.current_stream()
+                        # Sync the current stream with the copy stream
+                        current_stream.wait_stream(stream)
+                        # Ensure tensor memory is not reused until work on
+                        # main stream is complete
+                        output.record_stream(current_stream)  # type: ignore[arg-type]
+                elif hasattr(torch, "xpu") and torch.xpu.is_available():
+                    with torch.xpu.stream(stream):
+                        output = obj.to(target_gpu)
+                    # synchronize with the copy stream
+                    with torch.xpu.device(target_gpu):
```

Comment:
Looks like the code / logic is actually the same here, but instead of `torch.cuda.device` we have `torch.xpu.device` - is there any way to reduce the code duplication here?